pkgs/unified_analytics/lib/src/log_handler.dart

```diff
@@ -272,10 +272,18 @@ class LogHandler {
   /// This will keep the max number of records limited to equal to
   /// or less than [kLogFileLength] records.
   void save({required Map<String, Object?> data}) {
-    var records = logFile.readAsLinesSync();
-    final content = '${jsonEncode(data)}\n';
-
     try {
+      final stat = logFile.statSync();
+      List<String> records;
+      if (stat.size > kMaxLogFileSize) {
+        logFile.deleteSync();
+        logFile.createSync();
+        records = [];
+      } else {
+        records = logFile.readAsLinesSync();
+      }
```

Comment:
For simplicity, consider
```suggestion
      final stat = logFile.statSync();
      if (stat.size > kMaxLogFileSize) {
        logFile.deleteSync();
        logFile.createSync();
      }
      List<String> records = logFile.readAsLinesSync();
```

Reply:
This will incur an additional file disk read

Reply:
SGTM. The extra disk read would (_probably_) not happen very often, but I respect being cautious here.